### ISSUE TITLE
warnings from pry-debug with inspec shell under chef-workstation

### ISSUE THREAD

**Issue by clintoncwolfe** [inspec/inspec#5042]
# Describe the problem

Install the latest build of chef-workstation (available from Current channel on the downloads page). Run `eval "$(chef shell-init bash)"` to set your environment to run from Chef workstation. When you run `inspec shell`, you get:

```
[cwolfe@lodi temp]$ inspec shell
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-byebug-3.9.0/lib/pry-byebug/pry_ext.rb:13: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-0.13.1/lib/pry/core_extensions.rb:45: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'
Welcome to the interactive InSpec Shell
To find out how to use it, type: help

You are currently running on:

    Name:      mac_os_x
    Families:  darwin, bsd, unix, os
    Release:   18.7.0
    Arch:      x86_64

inspec> exit
[cwolfe@lodi temp]$ chef --version
Chef Workstation version: 20.5.45
Chef Infra Client version: 16.1.0
Chef InSpec version: 4.18.114
Chef CLI version: 3.0.1
Test Kitchen version: 2.5.1
Cookstyle version: 6.5.3
```

## Possible Solution

Aha! Link: https://chef.aha.io/features/SH-2176

**Comment by james-stocks** [comment]
This is already fixed upstream

```
jstocks@phobos inspec % ./inspec-bin/bin/inspec shell 
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-byebug-3.9.0/lib/pry-byebug/pry_ext.rb:13: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-0.13.1/lib/pry/core_extensions.rb:45: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'
Welcome to the interactive InSpec Shell
To find out how to use it, type: help

You are currently running on:

    Name:      mac_os_x
    Families:  darwin, bsd, unix, os
    Release:   19.4.0
    Arch:      x86_64

inspec> exit
jstocks@phobos inspec % chef gem install pry-stack_explorer
Fetching pry-stack_explorer-0.5.1.gem
Successfully installed pry-stack_explorer-0.5.1
1 gem installed
jstocks@phobos inspec % ./inspec-bin/bin/inspec shell      
Welcome to the interactive InSpec Shell
To find out how to use it, type: help

You are currently running on:

    Name:      mac_os_x
    Families:  darwin, bsd, unix, os
    Release:   19.4.0
    Arch:      x86_64

inspec> 

```

...however the [PR for the fix](https://github.com/pry/pry-stack_explorer/pull/49) indicates that pry-stack_explorer will no longer support Ruby 2.5 which may still be a problem for us. Or possibly not, if InSpec itself doesn't use pry-stack_explorer and workstation will only use ruby 2.7?

**Comment by clintoncwolfe** [comment]
OK, so I'm scanned the omnibus build of InSpec, and it does not include pry-stack_explorer. So we can drop #5043, and not worry about ruby 2.4-2.5 compatibility.

Meanwhile, that means we can use the released 0.5.1 gem, which only supports ruby 2.6 and 2.7 in Workstation, so https://github.com/chef/chef-workstation/pull/1225 can be modified to be a pin instead of using a fork.

One less fork...

**Comment by clintoncwolfe** [comment]
This is resolved in at least as of Chef Workstation 20.6.62 / inspec 4.19.0
